public/manifest.json

```diff
@@ -1,17 +1,34 @@
 {
-	"manifest_version": 2,
+	"manifest_version": 3,
 	"name": "Practice School Extender for BITS Pilani",
 	"version": "2.4.0",
 	"description": "Extends functionality of PSD interface by adding controls for sorting rows, backups, problem bank viewer",
-	"browser_action": {},
 	"icons": {
 		"32": "icon.png",
 		"48": "icon.png",
 		"64": "icon.png",
 		"128": "icon.png"
 	},
 	"background": {
-		"scripts": ["script.js"]
+		"service_worker": "script.js"
 	},
-	"permissions": ["activeTab"]
-}
+	"permissions": [
+		"activeTab",
+		"scripting"
+	],
+	"action": {},
```

Comment:
The `default_icon` key is recommended here (which is basically a copy-paste from the `icons` above).

Also, you can add a `default_title` here that is shown as a tooltip on hover:
```json
"action": {
		"default_icon": {
			"32": "icon.png",
			"48": "icon.png",
			"64": "icon.png",
			"128": "icon.png"
		},
		"default_title": "Click to run PS-Extender"
	}
```